src/clients/discord.ts

```diff
@@ -0,0 +1,131 @@
+const lastUserStateMsg = bridge.lastUserStateMsg;
+
+function registerDiscord(): void 
+{
+
+    discordClient.on('messageCreate', (m: Message<boolean>) =>
+    {
+        if(m.author.bot || m.guild === null)
+            return;
+
+        if(!targetDiscordChannel)
+            return;
+
+        if(m.channel.id !== targetDiscordChannel.id)
+            return;
+
+        //tmi.js automatically splits up these messages down if they are over 500 characters, so there's no need to worry if discord's message is too big.
+        const discordHeader: string = `[d][${ m.author.tag }] `,
+            foundIds: { [key: string]: boolean; } = {};
+
+        let finalMessage: string = m.content;
+
+        //Grab the contents of the message, convert discord mentions into usernames for twitch to see
+        for(const mention of m.content.matchAll(/<@[0-9]{1,}>/g))
+        {
+            if(mention === null)
+                continue;
+
+            const discordId: string = /[0-9]{1,}/.exec(mention[0])![0];
+
+            if(!foundIds[discordId])
+            {
+                foundIds[discordId] = true;
+                finalMessage = finalMessage.replaceAll(mention[0], '@[m]' + m.mentions.members!.get(discordId)!.user.tag);
+            }
+        }
+
+        //Also need to filter out custom emoji from discord; would be better to display the custom emoji name
+        for(const customEmoji of m.content.matchAll(/<:[A-Za-z]{1,}:[0-9]{1,}>/g))
+        {
+            const emojiNameAndId: string[] = /[A-Za-z]{1,}:[0-9]{1,}/.exec(customEmoji[0])![0].split(':');
+
+            if(!foundIds[emojiNameAndId[1]])
+            {
+                //Replace the ID of the custom emoji with just it's name
+                foundIds[emojiNameAndId[1]] = true;
+                finalMessage = finalMessage.replaceAll(customEmoji[0], '[e]' + emojiNameAndId[0]);
+            }
+        }
+
+
+        //Include attachments inside the message if they are present on discord
+        if(m.attachments?.size)
+            finalMessage += ' ' + [...m.attachments].map(e => e[1].url).join(' ');
+
+        const messageToSend: string = `${ discordHeader }${ finalMessage }`;
+
+        //Create a key-value pair that will be logged as a partially complete fused object. When we find the other piece on the twitch side, it will also be mapped in our collection.
+        const listNode: linkedListNode = messageLinkdListInterface.addNode(new conjoinedMsg(m));
+        twitchMessageSearchCache[messageToSend] = listNode;
+        discordTwitchCacheMap.set(m, listNode);
+
+        //I'm only grabbing the first index here... this will need to change if we scale up.
+        twitchClient.say(configFile.T2S_CHANNELS[0], messageToSend).then(undefined, genericPromiseError);
+
+        //Count upwards and delete the oldest message if need be
+        bridge.manageMsgCache();
+    });
+
+    const discordOnMesgDel = (m: Message<boolean> | PartialMessage): void =>
```

Comment:
Typescript requires the PartialMessage  as part of the type, due to messageDeleteBulk event handler recieving the messages collection w/ the type of Collection<string, Message<boolean> | PartialMessage>. Bad explanation but TLDR: required or else typescript no likey and throw error

Reply:
Yeah, just double checked Discord.JS's types; https://github.com/discordjs/discord.js/blob/main/packages/discord.js/typings/index.d.ts#L4228
It's of `[messages: Collection<Snowflake, Message | PartialMessage>, channel: GuildTextBasedChannel];` so we are required to have our collection be the same type of `Collection<Snowflake, Message | PartialMessage>`

Reply:
oof... that's annoying >_<
aight - sounds like it will need to be engraved in my head; or at least it will be littered everywhere - that will also help XP